services/admin/src/admin/routes/recreate_dataset.py

```diff
@@ -0,0 +1,92 @@
+from libcommon.exceptions import CustomError
+from libcommon.processing_graph import ProcessingGraph
+from libcommon.queue import Queue
+from libcommon.simple_cache import delete_dataset_responses
+from libcommon.storage import StrPath
+from libcommon.utils import Priority
+from libcommon.viewer_utils.asset import delete_asset_dir
+from starlette.requests import Request
+from starlette.responses import Response
+
+from admin.authentication import auth_check
+from admin.utils import (
+    Endpoint,
+    InvalidParameterError,
+    MissingRequiredParameterError,
+    UnexpectedError,
+    are_valid_parameters,
+    get_json_admin_error_response,
+    get_json_ok_response,
+)
+
+
+def create_recreate_dataset_endpoint(
+    processing_graph: ProcessingGraph,
+    assets_directory: StrPath,
+    cached_assets_directory: StrPath,
+    hf_endpoint: str,
+    hf_token: Optional[str] = None,
+    external_auth_url: Optional[str] = None,
+    organization: Optional[str] = None,
+    hf_timeout_seconds: Optional[float] = None,
+) -> Endpoint:
+    async def force_refresh_endpoint(request: Request) -> Response:
+        try:
+            dataset = request.query_params.get("dataset")
+            if not are_valid_parameters([dataset]) or not dataset:
+                raise MissingRequiredParameterError("Parameter 'dataset' is required")
+            try:
+                priority = Priority(request.query_params.get("priority", "low"))
+            except ValueError:
+                raise InvalidParameterError(
+                    f"Parameter 'priority' should be one of {', '.join(prio.value for prio in Priority)}."
+                )
+            logging.info(f"/recreate-dataset, dataset={dataset}, priority={priority}")
+
+            # if auth_check fails, it will raise an exception that will be caught below
+            auth_check(
+                external_auth_url=external_auth_url,
+                request=request,
+                organization=organization,
+                hf_timeout_seconds=hf_timeout_seconds,
+            )
+            # try to get the revision of the dataset (before deleting the jobs and the cache, in case it fails)
+            revision = get_dataset_git_revision(dataset=dataset, hf_endpoint=hf_endpoint, hf_token=hf_token)
+            # delete all the jobs and all the cache entries for the dataset (for all the revisions)
+            cancelled_jobs = Queue().cancel_dataset_jobs(dataset=dataset)
+            deleted_cached_responses = delete_dataset_responses(dataset=dataset)
+            if deleted_cached_responses is not None and deleted_cached_responses > 0:
+                # delete assets
+                delete_asset_dir(dataset=dataset, directory=assets_directory)
+                delete_asset_dir(dataset=dataset, directory=cached_assets_directory)
+            # add the first processing steps to the queue
+            for processing_step in processing_graph.get_first_processing_steps():
```

Comment:
Maybe try to use something similar to [try_backfill_dataset_then_raise](https://github.com/huggingface/datasets-server/blob/main/libs/libapi/src/libapi/utils.py#L109)?
What if the dataset does not exist, maybe we would be appending unnecessary jobs.
Not sure, feel free to ignore it if does not make sense.

Reply:
You're right! I forgot we add code for that (even if I wrote it). I made as in /webhook.